playbooks/adhoc/upgrades/upgrade.yml

```diff
@@ -1,4 +1,57 @@
 ---
+- name: Update deployment type
+  hosts: OSEv3
+  roles:
+  - openshift_facts
+  post_tasks: # technically tasks are run after roles, but post_tasks is a bit more explicit.
+  - openshift_facts:
+      role: common
+      local_facts:
+        deployment_type: "{{ deployment_type }}"
+
+- name: Verify upgrade can proceed
```

Comment:
This step should probably be before the Update deployment type step. Any pre-upgrade verification should be done prior to updating the deployment_type fact on the hosts.